### PR TITLE
pep8 and assorted improvements during code review

### PR DESCRIPTION
**Pull request by whoburg**

### CODE REVIEW

File: gpkit/geometric_program.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def __init__(self, cost, constraints, verbosity=1):
         self.cost = cost
         self.constraints = constraints
         self.posynomials = [cost] + list(constraints)
-        NomialData.__init__(self, signomials=self.posynomials, simplify=False)
+        super(GeometricProgram, self).__init__(signomials=self.posynomials,
```
Comment (bqpd) [inline gpkit/geometric_program.py]: This seems harder to read and less correct. We want to create a NomialData of the current object, not generic `super`

File: gpkit/nomials.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def __init__(self, exps=None, cs=1, require_positive=True, **descr):
             except AssertionError:
                 raise TypeError("cs and exps must have the same length.")
 
-        NomialData.__init__(self, exps, cs, simplify=simplify)
+        super(Signomial, self).__init__(exps, cs, simplify=simplify)
```
Comment (bqpd) [inline gpkit/nomials.py]: As above, I prefer calling `NomialData` explicitly.

**Comment by whoburg** [comment]
@bqpd do you want to work on the merge conflicts, or should I?

**Comment by bqpd** [comment]
These look mostly great! I can get the merge conflicts once we come to an agreement about `super` vs `NomialData`

File: gpkit/__init__.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def __enter__(self):
         global SIGNOMIALS_ENABLED
         SIGNOMIALS_ENABLED = True
 
-    def __exit__(self, type, value, traceback):
+    def __exit__(self, type_, val, traceback):
```
Comment (whoburg) [inline gpkit/__init__.py]: (this was to get pylint to stop complaining about overwriting `value` from the outer scope)

Comment (bqpd) [inline gpkit/__init__.py]: ah, nice.

**Comment by whoburg** [comment]
gotcha. I'm no expert, but I think calling `super` is certainly the preferred / correct way to implement inheritance. Is there a strong reason to deviate from that?

also, https://rhettinger.wordpress.com/2011/05/26/super-considered-super/

**Comment by whoburg** [comment]
said another way, should these perhaps not inherit from `NomialData`, but instead store a `NomialData` in a `data` attribute?

**Comment by bqpd** [comment]
It's great when: multi-inheritance, or changing inheritance.

But the idea here isn't quite regular inheritance it's just "superimpose this data structure on me".

**Comment by bqpd** [comment]
They could, but that's a bit inelegant and takes `exps` and `cs` another `.` away from the users.

**Comment by bqpd** [comment]
Merging this for now without super, we can move the discussion to #331

**Comment by whoburg** [comment]
Don't you mean the idea here isn't quite regular inheritance it's just "`super`impose this data structure on me" ? (couldn't resist)

**Comment by bqpd** [comment]
Pretty much :p

File: gpkit/geometric_program.py (hunk 1 above)
Comment (whoburg) [inline gpkit/geometric_program.py]: also, note this does create a NomialData of the current object -- that's why `self` is passed.

Comment (bqpd) [inline gpkit/geometric_program.py]: Yeah, they're identical. I guess I don't really care that much. I just feel like every time I see this I'm going to wonder "what the heck is it doing?"

Comment (whoburg) [inline gpkit/geometric_program.py]: yeah, Python's super calls are super clunky-looking.

Comment (bqpd) [inline gpkit/geometric_program.py]: using super()

File: gpkit/nomials.py (hunk 2 above)
Comment (bqpd) [inline gpkit/nomials.py]: in the end, using super()

**Pull request merged**